plugins/inputs/logfile/tailersrc.go

```diff
@@ -351,6 +352,27 @@ func (ts *tailerSrc) cleanUp() {
 			log.Printf("I! [logfile] Successfully removed file %v with auto_removal feature", ts.tailer.Filename)
 		}
 	}
+
+	// Ensure all buffers are flushed
+	if ts.buffer != nil {
```

Comment:
I'm curious why we are flushing logs here?

Reply:
removed changes here